Update Duration page (#4922)
jpmorganchase/salt-ds: site/docs/foundations/duration.mdx

  priority: 10
---

Duration represents the time between a user performing an action and the system giving feedback. It’s essential to indicate to users that something is happening when they interact with an element. Consistency in timing provides user comfort and can increase trust in a system. The duration should match the significance of the animated element or information.

> For detailed information on how to apply duration in design and the use cases, please refer to [this blog post](). (Internal users only)

Reviewer: Will tick once the link available

Reviewer 2: NIT blog post should be titled, rather than saying 'this blog post'.

> For detailed information on how to apply duration in design and the use cases, please refer to the [Understanding Duration in User Experience - Insights from the Salt Design System blog post](https://developer.test.aws.jpmchase.net/salt/news/blogs/20250402-duration#). (Internal users only)

Reviewer: ```suggestion
For detailed information on how to apply duration in design and the use cases, please refer to the [Understanding Duration in User Experience - Insights from the Salt Design System blog post](https://go/salt-ds-internal/salt/news/blogs/20250402-duration#). (Internal users only)
```

Duration represents the time between a user performing an action and the system giving feedback. It’s essential to indicate to users that something is happening when they interact with an element. Consistency in timing provides user comfort and can increase trust in a system. The duration should match the significance of the animated element or information.

> For detailed information on how to apply duration in design and the use cases, please refer to the [Understanding Duration in User Experience - Insights from the Salt Design System blog post](https://go/salt-ds-internal/salt/news/blogs/20250402-duration#). (Internal users only)

Reviewer: ```suggestion
For detailed information on how to apply duration in design and the use cases, please refer to the [Understanding Duration in User Experience - Insights from the Salt Design System blog post](https://go/salt-ds-internal/news/blogs/20250402-duration#). (Internal users only)
```

<Image
  src="/img/foundations/duration-instant.png"
  alt="Example of duration instant"
/>

## Perceptible (300 milliseconds)

A perceptible duration is a short detectable delay between a user's action and the system's response. It’s recommended for revealing secondary guidance or information to the user. For example, tooltips shouldn't appear instantly as the mouse moves over the UI.

<Image
  src="/img/foundations/duration-perceptible.png"
  alt="Example of duration perceptible"
/>

## Notable (1 second)

Notable delays are intentional and may be required to load or retrieve results. Should users experience delays longer than 1s, they may become frustrated, which can result in them abandoning the action they were trying to take. A 1s delay is usually the maximum limit for experiencing an interaction as one continuous flow. The user understands a ‘notable’ duration as a two-way interaction between them and the system.

> A progress cursor or [spinner component](go/salt/components/spinner) can be used to prevent users from wondering whether the action they just took didn’t work or is simply taking time to process.

Reviewer: Link from the same website, doesn't need `go` part, otherwise, public user won't be able to navigate successfully. Also, when go link is actually needed, prefix with `https://`

Reviewer 2: currently a broken link in this form.

> A progress cursor or [spinner component](https://www.saltdesignsystem.com/salt/components/spinner/examples) can be used to prevent users from wondering whether the action they just took didn’t work or is simply taking time to process.

Reviewer: ```suggestion
A progress cursor or [spinner component](/salt/components/spinner/examples) can be used to prevent users from wondering whether the action they just took didn’t work or is simply taking time to process.
```
